openpype/hosts/houdini/api/shelves.py

```diff
@@ -0,0 +1,196 @@
+                )
+                return
+
+            shelf = get_or_create_shelf(shelf_name)
+
+            if not shelf_definition.get('tools_list'):
+                log.warning("TOOLS INFO: No tool definition found for \
+shelf named {}".format(shelf_name))
+                return
+
+            mandatory_attributes = {'name', 'script'}
+            for tool_definition in shelf_definition.get('tools_list'):
+                # We verify that the name and script attibutes of the tool
+                # are set
+                if not all(
+                    tool_definition[key] for key in mandatory_attributes
+                ):
+                    log.warning("TOOLS ERROR: You need to specify at least \
+the name and the script path of the tool.")
+                    continue
+
+                tool = get_or_create_tool(tool_definition, shelf)
+
+                if not tool:
+                    return
+
+                # Add the tool to the shelf if not already in it
+                if tool not in shelf.tools():
+                    shelf.setTools(list(shelf.tools()) + [tool])
+
+            # Add the shelf in the shelf set if not already in it
+            if shelf not in shelf_set.shelves():
+                shelf_set.setShelves(shelf_set.shelves() + (shelf,))
+
+
+def get_or_create_shelf_set(shelf_set_label):
+    """This function verifies if the shelf set label exists. If not,
+    creates a new shelf set.
+
+    Arguments:
+        shelf_set_label {str} -- The label of the shelf set
+
+    Returns:
+        hou.ShelfSet -- The shelf set existing or the new one
+    """
+    all_shelves_sets = hou.shelves.shelfSets().values()
+
+    shelf_sets = [
+        shelf for shelf in all_shelves_sets if shelf.label() == shelf_set_label
+    ]
+
+    if shelf_sets:
+        return shelf_sets[0]
+
+    shelf_set_name = shelf_set_label.replace(' ', '_').lower()
+    new_shelf_set = hou.shelves.newShelfSet(
+        name=shelf_set_name,
+        label=shelf_set_label
+    )
+    return new_shelf_set
+
+
+def get_or_create_shelf(shelf_label):
+    """This function verifies if the shelf label exists. If not, creates
+    a new shelf.
+
+    Arguments:
+        shelf_label {str} -- The label of the shelf
+
+    Returns:
+        hou.Shelf -- The shelf existing or the new one
+    """
+    all_shelves = hou.shelves.shelves().values()
+
+    shelf = [s for s in all_shelves if s.label() == shelf_label]
+
+    if shelf:
+        return shelf[0]
+
+    shelf_name = shelf_label.replace(' ', '_').lower()
+    new_shelf = hou.shelves.newShelf(
+        name=shelf_name,
+        label=shelf_label
+    )
+    return new_shelf
+
+
+def get_or_create_tool(tool_definition, shelf):
+    """This function verifies if the tool exsist and update it. If not, creates
+    a new one.
+
+    Arguments:
+        tool_definition {dict} -- Dict with label, script, icon and help
+        shelf {hou.Shelf} -- The parent shelf of the tool
```

Comment:
```suggestion
        tool_definition (dict): Dict with label, script, icon and help
        shelf (hou.Shelf): The parent shelf of the tool
```

Shouldn't the type be inside `()` instead of `{}`?
And shouldn't it be followed by `: ` instead ` -- `

Reply:
You are right, I used autogenerated docstrings but I changed to match OpenPype's